components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-step1.jsp

```diff
@@ -245,29 +242,19 @@
             } else {
                 <%
                 String pattern = "";
-                ClaimDataAdminClient claimDataAdminClient = new ClaimDataAdminClient((String)
+
+                if(isAskPasswordEnabled) {
+                    ClaimDataAdminClient claimDataAdminClient = new ClaimDataAdminClient((String)
                 session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE),
                 CarbonUIUtil.getServerURL(config.getServletContext(), session),
                 (ConfigurationContext) config.getServletContext().getAttribute(CarbonConstants.CONFIGURATION_CONTEXT));
                 try {
-                    loop: for(LocalClaimDTO localClaim : claimDataAdminClient.getLocalClaims()){
-                        if (EMAIL_CLAIM.equals(localClaim.getLocalClaimURI())) {
-                            ClaimPropertyDTO[] claimPropertyDTOs = localClaim.getClaimProperties();
-                            if (claimPropertyDTOs != null) {
-                                for (ClaimPropertyDTO claimPropertyDTO : claimPropertyDTOs){
-                                    if (ClaimConstants.REGULAR_EXPRESSION_PROPERTY.equals(claimPropertyDTO.getPropertyName())) {
-                                        pattern = claimPropertyDTO.getPropertyValue();
-                                        pattern = "/" + pattern + "/";
-                                        break loop;
-                                    }
-                                }
-                            }
-                        }
-                    }
+                    pattern = claimDataAdminClient.getRegex(EMAIL_CLAIM);
                 }
-                catch (ClaimMetadataManagementServiceClaimMetadataException e) {
+                catch (Exception e) {
```

Comment:
Can't we catch only RemoteException & ClaimMetadataManagementServiceClaimMetadataException?

Reply:
No we cannot . Here it's fine to catch Exception. In a case if you introduce a new dedicated exception you can catch it. But definitely you can't catch ClaimMetadataManagementServiceClaimMetadataException here.